### PR TITLE
Internalize Newtonsoft.Json

### PR DESCRIPTION
**Pull request by gregkalapos**
With this PR we get rid of:
```
<PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
```
in `Elastic.Apm.csproj`. 

With this we avoid versions conflicts if the application or a dependency of the application depends on a different version of Newtonsoft.Json. 

TODO:
- [x] Add `NOTICE.txt`

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+==========
+Notice for: Elastic.Apm.Libraries.Newtonsoft.Json
```
Comment (russcam) [inline NOTICE.txt:2]:
```suggestion
Notice for: Elastic.Apm
```

File: NOTICE.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+----------
+Copy of the Newtonsoft.Json project version 13.0.1 by @JamesNK,
```
Comment (russcam) [inline NOTICE.txt:4]:
```suggestion
The source code in src/Elastic.Apm/Libraries/Newtonsoft.Json iscopied from Newtonsoft.Json by @JamesNK,
```

File: NOTICE.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+==========
```
Comment (russcam) [inline NOTICE.txt:1]: minor nitpick: I think we should name this file `NOTICE`, for consistency with the existing LICENSE.

**Review by russcam — COMMENTED** [review]
LGTM, left some comments around notice and wording

File: NOTICE.txt (hunk 1 above)
Comment (gregkalapos) [reply NOTICE.txt]: Done

File: NOTICE.txt (hunk 2 above)
Comment (gregkalapos) [reply NOTICE.txt]: Done

**Comment by gregkalapos** [comment]
I rebased this on master - there was 1 new code I needed to adapt [here](https://github.com/elastic/apm-agent-dotnet/pull/1241/commits/537e28568889fc4c0c261aa4f75c490ae75252da) - it's green now, merging this in.

**Pull request merged**